DiscordBot/Commands/EventCommands.cs

```diff
@@ -19,34 +20,109 @@ internal class EventCommands
+                    break;
+                case "edit":
+                    await context.RespondAsync("Not implemented yet!");
+                    break;
+            }
+        }
 
-            if (response == null)
+        public async Task CreateEvent(CommandContext context, InteractivityModule interactivity)
+        {
+            await context.RespondAsync($"{context.Member.Mention} - what is the name of the event?");
+            var eventName = await GetUserResponse(context, interactivity);
+            if (eventName == null)
             {
-                await context.RespondAsync($"{context.Member.Mention} - timed out");
+                return;
             }
-            else
+            
+            await context.RespondAsync($"{context.Member.Mention} - give an event description.");
+            var eventDescription = await GetUserResponse(context, interactivity);
+            if (eventDescription == null)
             {
-                await context.RespondAsync(response.Message.Content);
+                return;
             }
+
+            DateTimeOffset eventTime;
+
+            while (true)
+            {
+                await context.RespondAsync($"{context.Member.Mention} - what time is your event?");
+                var eventTimeResponse = await GetUserResponse(context, interactivity);
+                if (eventTimeResponse == null)
+                {
+                    return;
+                }
+
+                if (DateTimeOffset.TryParse(eventTimeResponse, out eventTime))
+                {
+                    break;
+                }
+            }
+
+            var embed = new DiscordEmbedBuilder
+            {
+                Title = eventName,
+                Description = eventDescription,
+                Color = new DiscordColor(0xFFFFFF),
+                Timestamp = eventTime
+            };
+
+            await context.RespondAsync(embed: embed);
+        }
+
+        private static async Task<string?> GetUserResponse(CommandContext context, InteractivityModule interactivity,
+            string[]? validStrings = null)
+        {
+            var response = interactivity.WaitForMessageAsync(
+                message =>
+                    IsValidResponse(message, context, validStrings),
+                TimeSpan.FromSeconds(30)
+            ).Result?.Message.Content;
+
+            switch (response)
+            {
+                case "stop":
+                    await context.RespondAsync($"{context.User.Mention} - operation stopped.");
+                    return null;
+                case null:
+                    await context.RespondAsync($"{context.User.Mention} - timed out.");
+                    break;
+            }
+
+            return response;
         }
 
-        private bool IsValidResponse(string response)
+        private static bool IsValidResponse(DiscordMessage response, CommandContext context, string[]? validStrings)
```

Comment:
Why does this need to be static? 🙂

Reply:
ReSharper gave me a squiggly line telling me to convert it to static, is there some convention for setting methods as static?

Reply:
Ah I was going to say it shouldn't if you access the valid strings as a private variable, but I realise now that you don't want to always pass them in. And actually that variable is static anyway so that shouldn't matter! So ignore me, feel free to leave 🙂